yb_build_thirdparty_main.py

```diff
@@ -958,6 +958,99 @@ def get_openssl_related_cmake_args(self):
+
+    # overridden in platform specific classes
+    def good_libs(self, file_path):
+        pass
+
+    def run(self):
+        log("Scanning installed executables and libraries...")
+        test_pass = True
+        # files to examine are much reduced if we look only at bin and lib directories
+        dir_pattern = '^(lib|libcxx|[s]bin)$'
+        dirs = [os.path.join(self.tp_installed_dir, type) for type in BUILD_TYPES]
+        for installed_dir in dirs:
+            with os.scandir(installed_dir) as candidate_dirs:
+                for candidate in candidate_dirs:
+                    if re.match(dir_pattern, candidate.name):
+                        examine_path = os.path.join(installed_dir, candidate.name)
+                        for dirpath, dirnames, files in os.walk(examine_path):
+                            for file_name in files:
+                                full_path = os.path.join(dirpath, file_name)
+                                if os.path.islink(full_path):
+                                    continue
+                                if not self.good_libs(full_path):
+                                    test_pass = False
+        if not test_pass:
+            fatal(f"Found problematic library dependencies, using tool: {self.tool}")
+        else:
+            log("No problems found with library dependencies.")
+
+
+class LibTestMac(LibTestBase):
+    def __init__(self):
+        super().__init__()
+        self.tool = "otool -L"
+        path_regex = ["^\t/usr/",
+                       "^\t/System/Library/",
+                       "^\t@rpath",
+                       "^\t@loader_path",
+                       f"^\t{self.tp_dir}"]
+        self.okay_paths = "|".join(path_regex)
+
+    def good_libs(self, file_path):
+        libout = subprocess.check_output(['otool', '-L', file_path]).decode('utf-8')
+        if 'is not an object file' in libout:
+            return True
+        return self.check_lib_deps(file_path, libout)
+
+
+class LibTestLinux(LibTestBase):
+    def __init__(self):
+        super().__init__()
+        self.tool = "ldd"
+        path_regex = ["^\tlinux-vdso",
+                       "^\t/lib64/",
+                       "^\t/opt/yb-build/brew/linuxbrew",
+                       "^\tstatically linked",
+                       "^.* => /lib64/",
+                       "^.* => /opt/yb-build/brew/linuxbrew",
+                       f"^.* => {self.tp_dir}"]
+        self.okay_paths = "|".join(path_regex)
+
+    def good_libs(self, file_path):
+        try:
+            # Hide messages about execute permissions
+            libout = subprocess.check_output(['ldd', file_path],
+                                             stderr=subprocess.DEVNULL).decode('utf-8')
+        except subprocess.CalledProcessError as cpe:
+            # Normal error code if file is not a shared library or executable
+            if cpe.returncode != 1:
+                log("Unexpected error for: ldd", file_path)
```

Comment:
Include the error code in the log message. Also I think file_path won't be logged properly this way.
Has to be something like:
```python
log("Unexpected exit code %d from ldd, file %s", cpe.returncode, file_path)
```